### DISCUSSION TITLE
Can't get auto instrumentation done with Play framework java

### DISCUSSION THREAD

**Discussion by protondrift** [open-telemetry/opentelemetry-java-instrumentation#2846] [Q&A] [answered]
Hi,
I had an Application wrote by JAVA and Playframework, Now I want to settle auto instrumentation to send trace to our trace backend via Opentelemetry.
I've added `opentelemetry-javaagent-all.jar` agent and some envs list below:
```
        - name: JVM_OPTS
          value: -J-Xms512m -J-Xmx2048m -J-javaagent:/server/opentelemetry-javaagent-all.jar
            -J-server
        - name: OTEL_TRACES_EXPORTER
          value: zipkin
        - name: OTEL_LOG_LEVEL
          value: debug
        - name: OTEL_METRICS_EXPORTER
          value: none
        - name: OTEL_EXPORTER_ZIPKIN_ENDPOINT
          value: http://simplest-collector.opentelemetry-operator-system:9411/api/v2/spans
```

logback.xml:
```
<root level="DEBUG">
```

After app startup, I didn't get a trace from opentelemetry, So I tried to debug with logging exporter 
```
        - name: OTEL_TRACES_EXPORTER
          value: logging
```

After restart the app, I didn't see any logs related to opentelemetry, Please tell me if some settings are missing by myself, Or how to debug the agent status, It has taken me a few days, Any comments will be appreciated.

My app's Dockerfile:
```
FROM openjdk:8u181-jdk

# Copy Code
ADD target/universal/app-1.0.0.tgz /server/
ADD conf/opentelemetry-javaagent-all.jar /server/

EXPOSE 9000
CMD bash bin/app ${JAVA_OPTS} -Dlogger.file=/server/config/logback.xml -Dconfig.file=/server/config/application.conf -Dhttp.port=9000 -Denv=${env}

```

```
agent version: 1.1.0
Play version: 2.6.20
```

**Comment by mateuszrzeszutek** [comment, marked as answer]
Are there no javaagent logs at all? Not even the `VersionLogger` startup log?

Your dockerfile seems to use `JAVA_OPTS` env var while the yaml fragment that you posted defines `JVM_OPTS` - is this correct? Does `JVM_OPTS` get translated to `JAVA_OPTS` somewhere? Also, there are those `-J` prefixes in the `JVM_OPTS` env var, are these stripped by some script?

Also, I don't think that `OTEL_LOG_LEVEL` is used by anything, if you want to enable the javaagent debug mode try `OTEL_JAVAAGENT_DEBUG = true`.

**Reply by protondrift** [reply]
Thx for the reply!

`Are there no javaagent logs at all? Not even the VersionLogger startup log?`
Yes, None related logs are showing, Just looks like classes aren't load.

```
Your dockerfile seems to use JAVA_OPTS env var while the yaml fragment that you posted defines JVM_OPTS - is this correct? Does JVM_OPTS get translated to JAVA_OPTS somewhere? Also, there are those -J prefixes in the JVM_OPTS env var, are these stripped by some script?
```

Thx for point this out, I think this is the root cause, After pass option I got this error:
`Unrecognized option: -J-javaagent:/server/opentelemetry-javaagent-all.jar`

I will try to add this option to my dockerfile and rebuild the image to see if it's working, List below To answer your question about `-J` prefix.
```
###  ------------------------------- ###
###  Start of customized settings    ###
###  ------------------------------- ###
usage() {
 cat <<EOM
Usage: $script_name [options]

  # jvm options and output control
  JAVA_OPTS          environment variable, if unset uses "$java_opts"
  -Dkey=val          pass -Dkey=val directly to the java runtime
  -J-X               pass option -X directly to the java runtime
                     (-J is stripped)
```

**Reply by protondrift** [reply]
Now it works, Thank you Mateusz! @mateuszrzeszutek
